src/main/java/com/ss/mqtt/broker/model/topic/TopicFilter.java

```diff
@@ -0,0 +1,39 @@
+package com.ss.mqtt.broker.model.topic;
+
+import org.jetbrains.annotations.NotNull;
+
+public class TopicFilter extends AbstractTopic {
+
+    public TopicFilter(@NotNull String topicName) {
+        super(topicName);
+
+        int multiPos = topicName.indexOf(MULTI_LEVEL_WILDCARD);
+        if (multiPos != -1 && multiPos != topicName.length() - 1) {
```

Comment:
what about endWith()?

Reply:
We should check if and only if MULTI_LEVEL_WILDCARD is placed on the end position. So endWith is not suitable here.